msbuild/Xamarin.Shared/Xamarin.Shared.targets

```diff
@@ -595,10 +596,62 @@ Copyright (C) 2018 Microsoft. All rights reserved.
 		 to the CollectAppManifestsDependsOn property. -->
 	<Target Name="CollectAppManifests" DependsOnTargets="$(CollectAppManifestsDependsOn)" />
 
-	<!-- This target has no inputs, because it must still be run if there are no input app manifests, to set default values -->
+	<Target Name="_CompileAppManifestInputs"
+		Condition="'$(IsMacEnabled)' == 'true' Or '$(IsHotRestartBuild)' == 'true'"
+		>
+		<PropertyGroup>
+			<_CompileAppManifestInputsFile>$(DeviceSpecificIntermediateOutputPath)_CompileAppManifest.inputs</_CompileAppManifestInputsFile>
+			<_CompileAppManifestFontFilesToRegister>@(_FontFilesToRegister)</_CompileAppManifestFontFilesToRegister>
+			<_CompileAppManifestPartialAppManifest>@(PartialAppManifest)</_CompileAppManifestPartialAppManifest>
+		</PropertyGroup>
+		<ItemGroup>
+			<_CompileAppManifestInputLine Include="SessionId=$(BuildSessionId)" />
```

Comment:
I'm not sure we'd want the SessionId here, IIRC it changes between builds, and doesn't affect the actual output.